src/libraries/System.Formats.Tar/tests/TarReader/TarReader.File.Async.Tests.cs

```diff
@@ -125,5 +126,245 @@ public class TarReader_File_Async_Tests : TarReader_File_Tests_Async_Base
+        {
+            await using MemoryStream archiveStream = GetTarMemoryStream(CompressionMethod.Uncompressed, folderName, testCaseName);
+            await using TarReader reader = new TarReader(archiveStream);
+            TarEntry entry;
+            while ((entry = await reader.GetNextEntryAsync()) != null)
+            {
+                AssertExtensions.GreaterThan(entry.Checksum, 0);
+                AssertExtensions.GreaterThan((int)entry.Mode, 0);
+            }
+        }
+
+        [Theory]
+        [InlineData("pax-multi-hdrs")] // Multiple consecutive PAX metadata entries
+        [InlineData("gnu-multi-hdrs")] // Multiple consecutive GNU metadata entries
+        [InlineData("neg-size")] // Garbage chars
+        [InlineData("invalid-go17")] // Many octal fields are all zero chars
+        [InlineData("issue11169")] // Checksum with null in the middle
+        [InlineData("issue10968")] // Garbage chars
+        [InlineData("writer-big")] // The size field contains an euro char
+        public async Task Throw_ArchivesWithRandomCharsAsync(string testCaseName)
+        {
+            await using MemoryStream archiveStream = GetTarMemoryStream(CompressionMethod.Uncompressed, "golang_tar", testCaseName);
+            await using TarReader reader = new TarReader(archiveStream);
+            await Assert.ThrowsAsync<FormatException>(async () => await reader.GetNextEntryAsync());
+        }
+
+        [Fact]
+        public async Task GarbageEntryChecksumZeroReturnNullAsync()
+        {
+            await using MemoryStream archiveStream = GetTarMemoryStream(CompressionMethod.Uncompressed, "golang_tar", "issue12435");
+            await using TarReader reader = new TarReader(archiveStream);
+            Assert.Null(await reader.GetNextEntryAsync());
+        }
+
+        [Theory]
+        [InlineData("golang_tar", "gnu-nil-sparse-data")]
+        [InlineData("golang_tar", "gnu-nil-sparse-hole")]
+        [InlineData("golang_tar", "gnu-sparse-big")]
+        [InlineData("golang_tar", "sparse-formats")]
+        [InlineData("tar-rs", "sparse-1")]
+        [InlineData("tar-rs", "sparse")]
+        public async Task SparseEntryNotSupportedAsync(string testFolderName, string testCaseName)
+        {
+            // Currently sparse entries are not supported.
+
+            // There are PAX archives archives in the golang folder that have extended attributes for treating a regular file as a sparse file.
+            // Sparse entries were created for the GNU format, so they are very rare entry types which are excluded from this test method:
+            // pax-nil-sparse-data, pax-nil-sparse-hole, pax-sparse-big
+
+            await using MemoryStream archiveStream = GetTarMemoryStream(CompressionMethod.Uncompressed, testFolderName, testCaseName);
+            await using TarReader reader = new TarReader(archiveStream);
+            await Assert.ThrowsAsync<NotSupportedException>(async () => await reader.GetNextEntryAsync());
```

Comment:
Is there a way to workaround an archive with sparse files? say, we still want to traverse the rest of entries.

Reply:
Unfortunately no. Note that sparse files are a rare scenario, and from the beginning it was decided that archives containing these would not be supported for 7.0.